system/doc/efficiency_guide/maps.xml

```diff
@@ -0,0 +1,618 @@
+      <c>is_element/2</c>) are avoided, <seeerl
+      marker="stdlib:ordsets">ordsets</seeerl> could be a better
+      choice than <seeerl marker="stdlib:sets">sets</seeerl>.</p></item>
+
+      <item><p>If the elements of the set are integers in a fairly
+      compact range, the set could be represented as an integer where
+      each bit represents an element in the set. The union operation
+      is performed by <c>bor</c> and the intersection operation by
+      <c>band</c>.</p></item>
+    </list>
+  </section>
+
+  <section>
+    <title>How Maps are Implemented</title>
+    <p>Internally, maps have two distinct representations depending on
+    the number of elements in the map. The representation will change
+    when a map grows beyond 32 elements, or when it shrinks to 32
+    elements or less.</p>
+
+    <list type="bulleted">
+      <item>A map with at most 32 elements has a compact
+      representation, making it suitable as an alternative to
+      records.</item>
+      <item>A map with more than 32 elements is represented as a tree
+      that can be efficiently searched and updated regardless of how
+      many elements there are.</item>
+    </list>
+
+    <section>
+      <title>How Small Maps are Implemented</title>
+      <p>A small map looks like this inside the runtime system:</p>
+      <table align="left">
+        <row>
+          <cell align="center"><c>FLATMAP</c></cell>
+          <cell align="center"><em>N</em></cell>
+          <cell align="center"><em>Keys</em></cell>
+          <cell align="center"><em>Value1</em></cell>
+          <cell align="center"><em>. . .</em></cell>
+          <cell align="center"><em>ValueN</em></cell>
+        </row>
+        <tcaption>The representation of a small map</tcaption>
+      </table>
+
+      <taglist>
+        <tag><c>FLATMAP</c></tag>
+        <item>The tag for a small map (called <em>flat map</em> in
+        the source code for the runtime system).</item>
+
+        <tag>N</tag>
+        <item>The number of elements in the map.</item>
+
+        <tag>Keys</tag>
+        <item>A tuple with keys of the map: <c>{Key1,...,KeyN}</c>. The
+        keys are sorted.</item>
+
+        <tag>Value1</tag>
+        <item>The value corresponding to the first key in the key tuple.</item>
+
+        <tag>ValueN</tag>
+        <item>The value corresponding to the last key in the key tuple.</item>
+      </taglist>
+
+      <p>As an example, let us look at how the map <c>#{a => foo, z => bar}</c>
+      would be represented:</p>
+
+      <table align="left">
+        <row>
+          <cell align="center"><c>FLATMAP</c></cell>
+          <cell align="center"><em>2</em></cell>
+          <cell align="center"><em>{a,z}</em></cell>
+          <cell align="center"><em>foo</em></cell>
+          <cell align="center"><em>bar</em></cell>
+        </row>
+        <tcaption>#{a => foo, z => bar}</tcaption>
+      </table>
+
+      <p>Let us update the map: <c>M#{q => baz}</c>. The map will
```

Comment:
The technical writer document says that "will" should be avoided.

Reply:
I have a lot of "will". I will remove some where I'll think it makes sense.